promoted_python_delivery_client/client/two_arm_experiment.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+from promoted_python_delivery_client.model.cohort_arm import CohortArm
+from promoted_python_delivery_client.model.cohort_membership import CohortMembership
+
+
+class TwoArmExperiment:
+    def __init__(self,
+                 cohort_id: str,
+                 num_active_control_buckets: int,
+                 num_control_buckets: int,
+                 num_active_treatment_buckets: int,
+                 num_treatment_buckets: int) -> None:
+        if not cohort_id.strip():
+            raise ValueError("Cohort ID must be non-empty")
+
+        if num_control_buckets < 0:
+            raise ValueError("Control buckets must be positive")
+
+        if num_treatment_buckets < 0:
+            raise ValueError("Treatment buckets must be positive")
+
+        if num_active_control_buckets < 0 or num_active_control_buckets > num_control_buckets:
+            raise ValueError("Active control buckets must be between 0 and the total number of control buckets")
+
+        if num_active_treatment_buckets < 0 or num_active_treatment_buckets > num_treatment_buckets:
+            raise ValueError("Active treatment buckets must be between 0 and the total number of treatment buckets")
+
+        self.cohort_id = cohort_id
+        self.cohort_id_hash = hash(cohort_id)
+        self.num_active_control_buckets = num_active_control_buckets
+        self.num_control_buckets = num_control_buckets
+        self.num_active_treatment_buckets = num_active_treatment_buckets
+        self.num_treatment_buckets = num_treatment_buckets
+        self.num_total_buckets = num_treatment_buckets + num_control_buckets
+
+    def check_membership(self, user_id: str) -> Optional[CohortMembership]:
+        hash_code = _combine_hash(hash(user_id), self.cohort_id_hash)
```

Comment:
Side - Thinking about this, it might make sense to flip the params.  cohort_id_hash will remain pretty constant.  The second is more important for randomizing the hash.

Reply:
I have the same comment about that, which is, do we want to do the same thing in every SDK?